sdk/keyvault/Azure.Security.KeyVault.Secrets/Readme.md

```diff
@@ -0,0 +1,254 @@
+# Azure Key Vault Secret client library for .NET
+Azure Key Vault is a cloud service that provides a secure storage of secrets, such as passwords and database connection strings. 
+
+Secret client library allows you to securely store and control the access to tokens, passwords, API keys, and other secrets. This library offers operations to create, retrieve, update, delete, purge, backup, restore and list the secrets and its versions.
+
+[Source code][secrets_client_src] | Package (NuGet) (coming soon) | [API reference documentation] (coming soon) | [Product documentation][keyvault_docs] | [Samples] (coming soon)
+
+## Getting started
+
+### Install the package
+Install the Azure Key Vault client library for .NET with [NuGet][nuget]:
+
+```PowerShell
+Install-Package Azure.Security.KeyVault.Secrets -Version 1.0.0-preview.1
+```
+
+### Prerequisites
+* An [Azure subscription][azure_sub].
+* An existing Key Vault. If you need to create a Key Vault, you can use the Azure Portal or [Azure CLI][azure_cli].
+
+If you use the Azure CLI, replace `<your-resource-group-name>` and `<your-key-vault-name>` with your own, unique names:
+
+```PowerShell
+az keyvault create --resource-group <your-resource-group-name> --name <your-key-vault-name>
+```
+
+### Authenticate the client
+In order to interact with the Key Vault service, you'll need to create an instance of the [SecretClient][secret_client_class] class. You would need a **vault url** and **client secret credentials (client id, client secret, tenant id)** to instantiate a client object. 
+
+Client secret credential way of authentication is being used in this getting started section but you can find more ways to authenticate with [Azure identity][azure_identity].
+
+ #### Create/Get credentials
+Use the [Azure CLI][azure_cli] snippet below to create/get client secret credentials.
+
+ * Create a service principal and configure its access to Azure resources:
+    ```PowerShell
```

Comment:
Just a question - powershell is something that is most commonly used by .NET developers?

Reply:
Yes :)